src/biogeophys/BareGroundFluxesMod.F90

```diff
@@ -337,20 +349,42 @@ subroutine BareGroundFluxes(bounds, num_noexposedvegp, filter_noexposedvegp, &
 
             tstar = temp1(p)*dth(p)
             qstar = temp2(p)*dqh(p)
-            z0hg_patch(p) = z0mg_patch(p) / exp(params_inst%a_coef * (ustar(p) * z0mg_patch(p) / 1.5e-5_r8)**params_inst%a_exp)
+
+            select case (z0param_method)
+            case ('ZengWang2007')
+               z0hg_patch(p) = z0mg_patch(p) / exp(params_inst%a_coef * (ustar(p) * z0mg_patch(p) / nu_param)**params_inst%a_exp)
+            case ('Meier2022')
+
+               ! After Yang et al. (2008)
+               z0hg_patch(p) = 70._r8 * nu_param / ustar(p) * exp( -beta_param * ustar(p)**(0.5_r8) * (abs(tstar))**(0.25_r8))
+               
```

Comment:
Let's remove this commented out code. We can't maintain it when commented out. And without scientific effort we can't add it, as we don't know if it really offers scientific value.

Reply:
Removed